### PR TITLE
General flash message component

### PR DESCRIPTION
**Pull request by programmertoni**
Adds: 

- GeneralFlashMsg component
- This component is bound to Redux store
- The functionality is basically the same as in master branch

### CODE REVIEW

File: app/controllers/client_api/users/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
+module ClientApi
```
Comment (ZmagoD) [inline app/controllers/client_api/users/users_controller.rb]: remove this file

File: app/controllers/client_api/users_controller.rb [hunk 2]
```diff
@@ -1,14 +0,0 @@
-module ClientApi
-  class UsersController < ApplicationController
```
Comment (ZmagoD) [inline app/controllers/client_api/users_controller.rb]: why you removed this file?

File: app/javascript/packs/app/routes.js [hunk 3]
```diff
@@ -1,6 +1,16 @@
+export const ROOT_PATH = "/";
+
 // activities
 export const ACTIVITIES_PATH = "/client_api/activities";
 
+// settings
```
Comment (ZmagoD) [inline app/javascript/packs/app/routes.js]: do you need all this to display the flash message?

File: app/javascript/packs/shared/actions/UsersActions.js [hunk 4]
```diff
@@ -1,5 +1,19 @@
 import axios from "../../app/axios";
-import { CURRENT_USER_PATH } from "../../app/routes";
+
+import {
+  CHANGE_USER_FULL_NAME_PATH,
```
Comment (ZmagoD) [inline app/javascript/packs/shared/actions/UsersActions.js]: you need that for flash messages?

File: app/javascript/packs/shared/general_flash_msg/index.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+              aria-label="Close"
+            >
+              <span
+                aria-hidden="true"
+                onClick={() => this.props.closeFlashMsg()}
+              >
+                ×
+              </span>
+            </button>
+            <OkSign className={`glyphicon glyphicon-${icon}`} />
```
Comment (ZmagoD) [inline app/javascript/packs/shared/general_flash_msg/index.jsx]: you can use Glyphicon component from react-bootstrap

File: app/javascript/packs/src/settings/components/MainNav.jsx [hunk 6]
```diff
@@ -3,9 +3,19 @@ import { Route, Switch, Redirect } from "react-router-dom";
...
-import { SETTINGS_ACCOUNT_PROFILE, SETTINGS_TEAMS } from "../../../app/routes";
+import {
+  ROOT_PATH,
+  SETTINGS_PATH,
+  SETTINGS_ACCOUNT_PATH,
+  SETTINGS_TEAMS_PATH,
+  SETTINGS_ACCOUNT_PROFILE,
+  SETTINGS_TEAMS,
+  SETTINGS_ACCOUNT_PROFILE_PATH
+} from "../../../app/routes";
```
[the remark this answers is not in the document]
Comment (ZmagoD) [reply app/javascript/packs/src/settings/components/MainNav.jsx]: Is OK to have this routes in this PR

**Comment by Ducz0r** [comment]
Closing as there's new PR in town - #803.

**Pull request closed without merging**